raiden-ts/src/transport/epics/rooms.ts

```diff
@@ -257,31 +252,40 @@ export function matrixHandleInvitesEpic(
         member.userId === matrix.getUserId() && member.membership === 'invite',
     ),
     withLatestFrom(config$),
-    mergeMap(([{ event, member, matrix }, { httpTimeout }]) => {
-      const sender = event.getSender(),
-        senderPresence$ = latest$.pipe(
-          pluckDistinct('presences'),
-          map((presences) => Object.values(presences).find((p) => p.payload.userId === sender)),
-          filter(isntNil),
-          take(1),
-          // Don't wait more than some arbitrary time for this sender presence update to show
-          // up; completes without emitting anything otherwise, ending this pipeline.
-          // This also works as a filter to continue processing invites only for monitored
-          // users, as it'll complete without emitting if no MatrixPresenceUpdateAction is
-          // found for sender in time
-          takeUntil(timer(httpTimeout)),
-        );
-      return senderPresence$.pipe(map((senderPresence) => ({ matrix, member, senderPresence })));
-    }),
+    mergeMap(([{ event, member, matrix }, { httpTimeout }]) =>
+      latest$.pipe(
+        pluckDistinct('presences'),
+        map((presences) =>
+          Object.values(presences).find((p) => p.payload.userId === event.getSender()),
+        ),
+        filter(isntNil),
+        take(1),
+        // Don't wait more than some arbitrary time for this sender presence update to show
+        // up; completes without emitting anything otherwise, ending this pipeline.
+        // This also works as a filter to continue processing invites only for monitored
+        // users, as it'll complete without emitting if no MatrixPresenceUpdateAction is
+        // found for sender in time
+        timeout(httpTimeout),
+        map((senderPresence) => ({ matrix, member, senderPresence })),
+        // on timeout error, leave invited room
+        catchError(() =>
+          defer(async () => {
+            log.info("Leaving invited room because we don't know the inviter", event);
+            return matrix.leave(member.roomId);
+          }).pipe(
+            retryWhile(intervalFromConfig(config$), { onErrors: [429] }),
```

Comment:
More general: In most cases we only handle 429 here. Should we also handle 500 and network errors?

Reply:
Yes, Paul, I think it's worth it to unify these 429, 500 and `networkErrors` handling, I didn't want to do this in this PR, but I think it'd be easier than creating another, I'll push a commit soon.

Reply:
It's fine for me to do it in a follow-up.